lib/epochs.js

```diff
@@ -132,6 +144,37 @@ module.exports = function Epochs(ssb) {
       })
     },
 
+    /**  Gets all epochs leading up to the specified epoch, all the way back to the root.
+     * Does not include the specified epoch nor epochs in epoch forks.
+     */
+    getPredecessorEpochs(groupId, epochRootId, cb) {
+      if (cb === undefined)
+        return p(this.getPredecessorEpochs).call(this, groupId, epochRootId)
+
+      getEpochs(groupId, (err, epochs) => {
```

Comment:
there is a better way to do this

```js
reduceEpochs => reduce
reduce.graph.getHistory(epochRootId).map(id => reduce.graph.getNode(id))
```

Reply:
I know about the tools because I built them, there's a lot going on with `reduce`

Reply:
:fire: do this change for sure

Reply:
ah cool didn't know getHistory was a thing.

rewrote now.

for some reason copy pasting the info-getting part of the function (members and secret etc) from getEpochs didn't work so had to rewrite it slightly, but things seem to work now.